profiler/profile.go

```diff
@@ -175,6 +181,28 @@ var profileTypes = map[ProfileType]profileType{
 			return buf.Bytes(), err
 		},
 	},
+	executionTrace: {
+		Name:     "execution-trace",
+		Filename: "trace.gotrace", // TODO: pick a good name
+		Collect: func(p *profiler) ([]byte, error) {
+			if !p.shouldTrace() {
+				return nil, errors.New("execution trace collection is disabled")
+			}
+			defer func() {
+				// TODO: this updates the last trace time
+				// regardless of whether tracing suceeds or not.
+				// Should we only update it if tracing succeeds?
+				p.lastTrace = time.Now()
+			}()
+			buf := new(bytes.Buffer)
+			if err := trace.Start(buf); err != nil {
+				return nil, err
+			}
+			p.interruptibleSleep(p.cfg.traceConfig.Duration)
```

Comment:
I also considered putting a sleep _before_ starting the trace as well. We could collect the trace at a random point within the profiling period rather than always doing it at the start of the period.

Reply:
I'll leave a TODO and pick it up in a subsequent PR.